### PR TITLE
Deflake dex tests

### PR DESCRIPTION
**Pull request by actgardner**
Install dex on the TestFaster k8s clusters when we're running an auth test bucket.

### CODE REVIEW

File: Makefile
Comment (actgardner) [inline Makefile:387]: I'm not sure what the difference is between the version of helm in testfaster and the version in Travis - I think we're OK to drop this for now. Once the identity server alpha lands these will use the identity server instead of dex anyways.

**Review by lukemarsden — APPROVED** [review]
nice work :tada:! i didn't realise we were just failing to launch dex! sorry :)

as a slight optimization, you could move `make launch-dex` into https://github.com/pachyderm/pachyderm/blob/master/.testfaster.yml#L115 so that it gets started ahead of the tests running, which would speed things up a bit, but that has the possible downside of running dex in places it's not needed - so your call. feel free to merge as-is, or make that change and merge without further review once CI is green. thanks!

**Comment by lukemarsden** [comment]
Btw as long as you are reasonably confident changes aren't making Travis work, I don't mind if you merge deflaking changes with failing Travis builds. Insisting on Travis being green for every single change will slow us down.

**Pull request merged**
